### ISSUE TITLE
Interface Module Procedure not Set as Public

### ISSUE THREAD

**Issue by rosenbrockc** [rosenbrockc/fortpy#56]
When compiling the unit tests for `gcd_2ints` in `rational_mathematics.f90` of `msg-byu/symlib`, the following compilation error arises:

```
gcd_2ints.f90:35.20:

    gcd_2ints_fpy = gcd_2ints(x1, x2)
                    1
Error: Function 'gcd_2ints' at (1) has no IMPLICIT type
gfortran: error: gcd_2ints.o: No such file or directory
```

It is caused by the `gcd_2ints` subroutine not being made public by fortpy. The module is part of an interface, which is probably the cause of the problem. We decided that interface procedures should be tested independently since the ability of the interface to choose the right one is part of the compiler.

**Comment by rosenbrockc** [comment]
Fixed in Revision 1.6.2; the problem was that the `interface` keyword was written as `INTERFACE` in the code file and the logic to detect where the block of `public` statements ends was case sensitive.

**Comment by wsmorgan** [comment]
This error seems to have re-emerged in the ralloc and find_value_in_array interfaces in utilities.f90 and vector_matrix_utilities.f90 respectively. I've tried various ways of capitalizing and un-capitalizing the key words. The error appears as follows for both.
```
ERROR: grep: repetition-operator operand invalid

WARNING: compile generated some errors or warnings:
ralloc_iml.f90:19.35:

  ralloc_integer_matrix_list_fpy = ralloc_integer_matrix_list(p, n)
                                   1
Error: Function 'ralloc_integer_matrix_list' at (1) has no IMPLICIT type
gfortran: error: ralloc_iml.o: No such file or directory

ERROR: Could not compile executable ralloc_iml.x
```
